docs/developers_guide.rst

```diff
@@ -0,0 +1,115 @@
+ORA2 Developer Guide
+====================
+
+.. contents::
+
+Where is the code
+-----------------
+ORA is broken into two separate repositories:
+
+- `edx-ora2 <https://github.com/edx/edx-ora2>`_
+
+- `edx-submissions <https://github.com/edx/edx-submissions>`_
+
+Using ORA with docker devstack
+------------------------------
+Note - "from inside the lms", means you've run 'make lms-shell' from the devstack directory and are on a command prompt inside the lms container.
+
+1. Clone edx-ora2 repo into ../src/ directory (relative to your 'devstack' repo location). 
+
+2. From the src directory, run the following command to install ora2 in the lms's and studio's virtual environments:
+
+ ``make install-local-ora``
+
+Alternatively, one can log into either lms (``make lms-shell``) or studio (``make studio-shell``) and execute the following commands to uninstall ora2 and reinstall your local copy;
+ 
+ ``pip uninstall ora2 -y; pip install -e /edx/src/edx-ora2/``
+
+3. Now, get your edx-ora2 development environment set up: (the virtual environment MUST be named **edx-ora2**)
+This must be done from inside either lms or studio docker container.
+
+- ``cd /edx/src/edx-ora2``
+- ``virtualenv edx-ora2``
+- ``source edx-ora2/bin/activate``
+- ``make install``
+- ensure that your virtual environment is named **edx-ora2**. Using a different name will cause errors when trying to generate translations.
+
+4. Now, in the devstack directory on your host, run:
+``make lms-restart lms-logs``
+
+5. That's it, you're good to go! See Makefile for all the available commands, most are fairly self-explanatory
+In order to simulate a given tox environment (django18, django111, quality, js), run tox -e <env> for the env in question (after re-activating your edx-ora2 virtual environment).
+Usually, you can just run the underlying make commands for quicker tests, as requirements aren't re-installed.
+
+Working with submission dependencies
+------------------------------------
+ORA code also depends on edx-submissions. As a result, anytime a new version of edx-submissions is released, ORA code must be updated as follows.
+
+For devstack:
+
+- ``make lms-shell``   # get into docker container
+
+- ``cd /edx/src/edx-ora2`` # goto into ora2 folder
+
+- ``source edx-ora2/bin/activate`` # activate ora2 virtual environment
+
+- ``pip install -U edx-submissions`` # installs latest version of submissions from pypi
+
+          OR
+          
+- ``pip install -e /edx/src/edx-submissions`` # istalls local version of submissions
```

Comment:
Just a note that this renders strangely in GitHub's preview. Consider reformatting.